common/auto-updater/versions-db-manager.cpp

```diff
@@ -0,0 +1,307 @@
+    {
+        // Load server versions info on first access
+        if (!init())
+        {
+            out_version = 0;
+            return false;
+        }
+
+        auto platform = get_platform();
+
+        std::string up_str(to_string(up));
+        std::string comp_str(to_string(cp));
+
+        if (up_str.empty() || comp_str.empty()) return false;
+
+        // Look for the required version
+        auto res = std::find_if(_server_versions_vec.begin(), _server_versions_vec.end(),
+            [&, dev_name, up_str, comp_str, platform](std::unordered_map<std::string, std::string> ver)
+        {
+            return (dev_name == ver["device_name"] && up_str == ver["policy_type"] && comp_str == ver["component"] && (platform == ver["platform"] || ver["platform"] == "*"));
+        });
+
+        if (res != _server_versions_vec.end())
+        {
+            auto version_str = (*res)["version"];
+            out_version = version(version_str);
+            return true;
+        }
+
+        out_version = 0;
+        return false; // Nothing found
+    }
+
+    bool versions_db_manager::get_ver_data(const component_part_type cp, const version& version, const std::string& req_field, std::string& out)
+    {
+        // Load server versions info on first access
+        if (!init()) return false;
+
+        std::string platform = get_platform();
+
+        std::string comp_str(to_string(cp));
+
+        if (comp_str.empty()) return false;
+
+        // Look for the required version
+        auto res = std::find_if(_server_versions_vec.begin(), _server_versions_vec.end(),
+            [this, version, comp_str, platform](std::unordered_map<std::string, std::string> ver)
+        {
+            return (versions_db_manager::version(ver["version"]) == version && comp_str == ver["component"] && (platform == ver["platform"] || ver["platform"] == "*"));
+        });
+
+        if (res != _server_versions_vec.end())
+        {
+            out = (*res)[req_field];
+            return true;
+        }
+
+        return false; // Nothing found
+    }
+
+    std::string versions_db_manager::convert_version_to_formatted_string(const long long ver_num) const
+    {
+        std::stringstream raw_ver_str, fixed_ver_str;
+        raw_ver_str << std::setfill('0') << std::setw(10) << ver_num;
+        fixed_ver_str << raw_ver_str.str().substr(0, 2) << "."
+            << raw_ver_str.str().substr(2, 2) << "."
+            << raw_ver_str.str().substr(4, 2) << "."
+            << raw_ver_str.str().substr(6, 4);
+        return fixed_ver_str.str();
+    }
+
+    std::string versions_db_manager::to_string(const component_part_type& comp) const
+    {
+        switch (comp)
+        {
+        case LIBREALSENSE: return "LIBREALSENSE";
+        case VIEWER: return "VIEWER";
+        case DEPTH_QUALITY_TOOL: return "DEPTH_QUALITY_TOOL";
+        case FIRMWARE: return "FIRMWARE";
+            break;
+        default:
+            break;
```

Comment:
Default case underlines invalid input - pls generate error log with the proper value for debugging

Reply:
Agreed

Reply:
Done